### PR TITLE
Add a checker to ensure that the currently installed version of Plesk is available on autoinstall.plesk.com

### PR DESCRIPTION
**Pull request by SandakovMM**

### CODE REVIEW

**Review by kpushkaryov — CHANGES_REQUESTED** [review]

File: pleskdistup/common/src/version.py [hunk 1]
```diff
@@ -151,3 +151,61 @@ def __eq__(self, other) -> bool:
 
     def __ge__(self, other) -> bool:
         return not self.__lt__(other)
+
+
+class PleskVersion():
```
[the remark this answers is not in the document]
Comment (SandakovMM) [reply pleskdistup/common/src/version.py:156]: Let's add some additional commentary

[the remark this answers is not in the document]
Comment (kpushkaryov) [reply pleskdistup/common/src/version.py:156]: I mean, you don't need `()`.

File: pleskdistup/common/src/version.py [hunk 2]
```diff
@@ -151,3 +151,61 @@ def __eq__(self, other) -> bool:
...
+        self._extract_from_version(version)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(major={self.major!r}, minor={self.minor!r}, patch={self.patch!r}, hotfix={self.hotfix!r})"
+
+    def __str__(self) -> str:
+        return f"{self.major}.{self.minor}.{self.patch}.{self.hotfix}"
+
+    def __lt__(self, other) -> bool:
+        return (self.major, self.minor, self.patch, self.hotfix) < (other.major, other.minor, other.patch, other.hotfix)
```
Comment (kpushkaryov) [inline pleskdistup/common/src/version.py:205]: Repeating `(major, minor, ...)` seems excessive, please create a private method converting the version to a tuple.

File: pleskdistup/common/src/plesk.py [hunk 3]
```diff
@@ -5,9 +5,14 @@
 import re
 import subprocess
 import typing
+import urllib.request
+import xml.etree.ElementTree as ElementTree
 
-from . import log, mariadb, systemd
+from . import log, mariadb, systemd, version
 
+# http://autoinstall.plesk.com/products.inf3 is a xml file with available products,
```
Comment (kpushkaryov) [inline pleskdistup/common/src/plesk.py:13]: a xml -> an xml

File: pleskdistup/common/src/plesk.py [hunk 4]
```diff
@@ -5,9 +5,14 @@
...
+# including all versions of plesk.
```
Comment (kpushkaryov) [inline pleskdistup/common/src/plesk.py:14]: plesk -> Plesk

**Review by kpushkaryov — CHANGES_REQUESTED** [review]

**Review by kpushkaryov — APPROVED** [review]
Please rebase before merge.

**Pull request merged**